### DISCUSSION TITLE
Docker installation is slow on Mac

### DISCUSSION THREAD

**Discussion by MathewSalazarTS** [pimcore/pimcore#17399] [Q&A / Support]
Has anyone experienced this? 

Any tips on what I can do?

**Comment by kingjia90** [comment]
Unfortunately i am not able to help in this topic (i don't have a Mac 😄 ), but i have found some helpful past discussion, please have a look at  this
https://github.com/pimcore/docker/pull/76#issuecomment-1060882821

**Comment by markus-moser** [comment]
I'm working on Mac too and agree that there are some performance problems, especially for tasks where a lot of files are involved (e.g. composer update, symfony cache clear + warmup). I ended up using docker pro with the synchronized file shares feature (https://docs.docker.com/desktop/synchronized-file-sharing/). With this the performance is fine but there sometimes is some lag when changing files (because they need to be synched in the background). Not perfect, but quite ok. At the end the only OS where it works perfectly without downsides is linux. Its a bit annoying that in 2024 there is still no possibility to have an OS as developer without some major downsides. I would need a mixture of linux, windows and mac os 😄

**Reply by natted** [reply]
I believe mutagen provides the file sync feature you mention to improve speed. 
@MSTadashiShoji  Mutagen may be able to installed separately without the docker pro sub ( https://mutagen.io )
DDEV also had mutagen sync integration last I checked, which probably can be configured to support pimcore (and adds other nice things for local development)

**Reply by dpfaffenbauer** [reply]
Use orbstack, much better and faster

**Reply by markus-moser** [reply]
@dpfaffenbauer Thanks, gave it a short try. Maybe I'm doing something wrong but for me orbstack doesn't really work well at the moment. I installed orbstack and migrated from docker desktop with the built-in assistent. 

Now for example a `composer dump` in my demo installation runs 3:18 min whereas on docker pro with the file sharing feature enabled it needs 10 seconds.

**Reply by dpfaffenbauer** [reply]
runs for 5 seconds on a bigger project. ARM or Intel Mac?

**Reply by markus-moser** [reply]
ARM (M2 Pro)

**Reply by MathewSalazarTS** [reply]
@dpfaffenbauer I've tried orbstack and unfortunately my results are the same.

For everyone, this is probably off track but I've just been looking at the execution timeline provided by Symfony. I wanted to make sure that Docker's read/write with the volume mount is the problem in this case.  Since the controller is taking up some time, I am assuming its Docker. Right? 

![Screenshot 2024-07-26 at 11 47 08 AM](https://github.com/user-attachments/assets/4803f470-4259-4853-84ae-6306a710d885)

FYI: Running on ARM (M1 Pro) and Everyone is awesome for the quick responses! Thank you!
